packages/react/src/components/OverflowSet/OverflowButton.tsx

```diff
@@ -4,26 +4,51 @@ import { IKeytipProps } from '../../Keytip';
 import { IOverflowSetItemProps, IOverflowSetProps } from './OverflowSet.types';
 import { useConst } from '@fluentui/react-hooks';
 
+const unregisteredKeytipID = 'keytipToBeRegistered';
+
+const keytipHasBeenRegistered = (id: string) => {
+  return id.indexOf(unregisteredKeytipID) === -1;
+};
+
+const registerPersistedKeytip = (
+  keytip: IKeytipProps,
+  keytipManager: KeytipManager,
+  persistedKeytips: { [uniqueID: string]: IKeytipProps },
+  oldKeytipId: string,
+) => {
+  const uniqueID = keytipManager.register(keytip, true);
+  // Update map
+  persistedKeytips[uniqueID] = keytip;
+  delete persistedKeytips[oldKeytipId];
+};
+
 const useKeytipRegistrations = (
   persistedKeytips: { [uniqueID: string]: IKeytipProps },
   keytipManager: KeytipManager,
 ) => {
-  React.useEffect(() => {
+  React.useLayoutEffect(() => {
     Object.keys(persistedKeytips).forEach((keytipId: string) => {
       const keytip = persistedKeytips[keytipId];
-      const uniqueID = keytipManager.register(keytip!, true);
-      // Update map
-      persistedKeytips[uniqueID] = keytip;
-      delete persistedKeytips[keytipId];
+      if (keytipHasBeenRegistered(keytipId)) {
+        // Keytip has been registered, update
+        keytipManager.update(keytip, keytipId);
+      } else {
+        // Keytip hasn't been registered, register it
+        registerPersistedKeytip(keytip, keytipManager, persistedKeytips, keytipId);
+      }
     });
+  });
```

Comment:
Do you want this fn to run on ever render or should it run when persistedKeytips or keytipManager changes (like it did before)?

Reply:
the problem I was seeing was that persistedKeytips doesn't actually change, it's a dictionary so it's members change but the object is a constant. At least, that's how I understood it working. this pattern of code emulates the old code in the sense that yes it runs a lot

Reply:
Hmm, I would expect it to be different every single time (and maybe you would even need to use `useRef` to make it not over render)... Maybe @czearing @khmakoto  or @behowell have some more insight here

Reply:
yeah this handled better in useKeytipData with usePrevious because it's looking at a single entity. I'll look into this + your other comments more today

Reply:
I tried as much as possible to closely follow what useKeytipData does. the whole point is that keytips should mount, update, and unmount when their respective components mount, update, and unmount. The difference for OverflowButton is that it's in charge of the persisted keytips so there's not a direct correlation

The previous (v7) code worked like this, where it registered and unregistered the persisted keytips on mount, update, and unmount. There currently is no mechanism for updating persisted keytips, I looked into adding that but it was adding more unknowns and complexity for this change. Persisted keytips are also a small subset of keytips as a whole so having them update a few extra times is something we could revisit later